tests/StackExchange.Utils.Tests/HttpTests.cs

```diff
@@ -221,14 +222,60 @@ public async Task PatchRequest()
             Assert.Equal(Http.DefaultSettings.UserAgent, result.Data.Headers["User-Agent"]);
         }
 
+        [Fact]
+        public async Task SendFormUsesMultipartFormData()
+        {
+            string expectedFormKey = "key";
+            string expectedFormValue = "value";
+            var form = new NameValueCollection
+            {
+                [expectedFormKey] = expectedFormValue
+            };
+            var result = await Http.Request("https://httpbin.org/post")
+                .SendForm(form)
+                .ExpectJson<HttpBinResponse>()
+                .PostAsync();
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+            var actualContentType = Assert.Contains("Content-Type", (IDictionary<string, string>)result.Data.Headers);
+            Assert.StartsWith("multipart/form-data;", actualContentType);
+            var actualFormValue = Assert.Contains(expectedFormKey, (IDictionary<string, string>)result.Data.Form);
+            Assert.Equal(expectedFormValue, actualFormValue);
```

Comment:
`Assert.Contains` has ambiguous overloads for these args (one taking `IDictionary<,>` and another taking `IReadOnlyDictionary<,>`), hence the casting here.

I could remove casts if that's easier to read:

```suggestion
            Assert.True(result.Data.Headers.ContainsKey("Content-Type"));
            Assert.StartsWith("multipart/form-data;", result.Data.Headers["Content-Type"]);
            Assert.True(result.Data.Form.ContainsKey(expectedFormKey));
            Assert.Equal(expectedFormValue, result.Data.Form[expectedFormKey]);
```

Or just throw like the other tests if those aren't on the response:

```suggestion
            Assert.StartsWith("multipart/form-data;", result.Data.Headers["Content-Type"]);
            Assert.Equal(expectedFormValue, result.Data.Form[expectedFormKey]);
```

Reply:
I think your first instinct here is good and easy to read, 👍